File: Sources/AnimalFarm/Animal/Animals.swift
//
//  Animals.swift
//  AnimalFarm
//
//  Created by Liam Rosenfeld on 1/30/19.
//  Copyright © 2019 Liam Rosenfeld. All rights reserved.
//

public let animals = [
    "bunny": bunny,
    
    "cow": Animal(
        art: #"""
        ^__^
        (oo)\_______
        (__)\ o O o )\/\
            ||----w |
            ||     ||
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "tux": Animal(
        art: #"""
            .--.
           |o_o |
           |:_/ |
          //   \ \
         (|     | )
        /'\_   _/`\
        \___):(___/
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "cat": Animal(
        art: #"""
            /\_/\           ___
           : o_o :_______    \ \
              ^      __(  \.__) )
        (@)<_____>__(_____)____/
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "dog": Animal(
        art: #"""
           __
        o-''|\_____/)
         \_/|_)     )
            \  __  /
            (_/ (_/
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "pig": Animal(
        art: #"""
            _____
        ^..^     \9
        (oo)_____/
           WW  WW
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "hedgehog": Animal(
        art: #"""
           ..::::::::.
          :::::::::::::
         /. `:::::::::::
        o__,_::::::::::'
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "dino": Animal(
        art: #"""
         __
        (_ \
          \ \_.----._
           \         \
           |  ) |  )  \__
           |_|--|_|'-.__/
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    ),

    "frog": Animal(
        art: #"""
              _  _
             (.)(.)
         ,-.(.____.),-.
        ( \ \ '--' / / )
         \ \ / ,. \ / /
          ) '| || |' (
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["say"]!
    )
]

internal let bunny = Animal(
        art: #"""
        (\__/) ||
        (•ㅅ•) ||
        / 　 づ
        """#,
        recBubbleWidth: 15,
        defaultBubble: bubbles["sign"]!
)

File: src/converter/animals.rs
pub enum Animal {
    Bunny,
    Cow,
    Tux,
    Cat,
    Dog,
    Pig,
    Hedgehog,
    Dino,
    Frog,
}

impl Animal {
    pub fn art(&self) -> &str {
        match *self {
            Animal::Bunny => BUNNY,
            Animal::Cow => COW,
            Animal::Tux => TUX,
            Animal::Cat => CAT,
            Animal::Dog => DOG,
            Animal::Pig => PIG,
            Animal::Hedgehog => HEDGEHOG,
            Animal::Dino => DINO,
            Animal::Frog => FROG,
        }
    }
}

const BUNNY: &str = r#"
(\__/) ||
(•ㅅ•) ||
/ 　 づ
"#;

const COW: &str = r#"
\  ^__^
 \ (oo)\_______
   (__)\ o O o )\/\
       ||----w |
       ||     ||
"#;

const TUX: &str = r#"
\   .--.
 \ |o_o |
   |:_/ |
  //   \ \
 (|     | )
/'\_   _/`\
\___):(___/
"#;

const CAT: &str = r#"
\   /\_/\           ___
 \ : o_o :_______    \ \
      ^      __(  \.__) )
(@)<_____>__(_____)____/
"#;

const DOG: &str = r#"
\     __
 \ o-''|\_____/)
    \_/|_)     )
        \  __  /
        (_/ (_/
"#;

const PIG: &str = r#"
\     _____
 \ ^..^     \9
   (oo)_____/
      WW  WW
"#;

const HEDGEHOG: &str = r#"
\      ..::::::::.
 \    :::::::::::::
  \  /. `:::::::::::
    o__,_::::::::::'
"#;

const DINO: &str = r#"
\   __
 \ (_ \
     \ \_.----._
      \         \
      |  ) |  )  \__
      |_|--|_|'-.__/
"#;

const FROG: &str = r#"
 \    _  _
  \  (.)(.)
 ,-.(.____.),-.
( \ \ '--' / / )
 \ \ / ,. \ / /
  ) '| || |' (
"#;
